src/de/grobox/liberario/Preferences.java

```diff
@@ -98,6 +105,44 @@ public static void setPref(Context context, String pref, boolean value) {
 		editor.commit();
 	}
 
+	public static EnumSet<Product> getProducts(Context cont) {
+		// check if there's a preference for selected products for the current network
+		TransportNetwork current_network = Preferences.getTransportNetwork(cont);
+		String network_pref = current_network.getId().toString() + "_selected_products";
+		SharedPreferences sp = cont.getSharedPreferences(Preferences.NETWORK_SETTINGS, Context.MODE_PRIVATE);
+		// get the selected products from the pref
+		Set<String> string_set = sp.getStringSet(network_pref, null);
+		// if there is a pref for the current network, read the selected products and update
+		if(string_set != null) {
+			EnumSet<Product> products = EnumSet.noneOf(Product.class);
+			for(String s : string_set) {
+				products.add(Product.valueOf(s));
+			}
+			Serializable tmpProducts = products;
+			products = (EnumSet<Product>) tmpProducts;
+			return products;
+		} else {
+			return EnumSet.noneOf(Product.class);
+		}
+	}
+
+	public static void setProducts(Context cont, EnumSet<Product> products) {
+		// get the current network
```

Comment:
Before this even happens, you could check if `products` is equal to *all* products and if so just return here without saving anything. (This would require returning all products above)